src/trustRegion.jl

```diff
@@ -80,10 +80,18 @@ for large-scale and numerically-difficult nonlinear systems.
     Currently, the linear solver and chunk size choice only applies to in-place defined
     `NonlinearProblem`s. That is expected to change in the future.
 """
-struct TrustRegion{CS, AD, FDT, L, P, ST, CJ, MTR} <:
+EnumX.@enumx RadiusUpdateSchemes begin
+    Simple
+    Hei
+    Yuan
+    Bastin
+end
+
+struct TrustRegion{CS, AD, FDT, L, P, ST, CJ, MTR, RUS} <:
        AbstractNewtonAlgorithm{CS, AD, FDT, ST, CJ}
     linsolve::L
     precs::P
+    radius_update_scheme::RUS
```

Comment:
```suggestion
    radius_update_scheme::RadiusUpdateSchemes.T
```

It doesn't need to be parametric.,